<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1175</width>
    <height>686</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>20</y>
      <width>20</width>
      <height>601</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>200</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Particle Count</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>260</y>
      <width>91</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>-9.8</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox_2">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>300</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Ground Plane</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox_4">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>160</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Cloth Simulation</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>470</y>
      <width>261</width>
      <height>151</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="pushButton_2">
       <property name="text">
        <string>Simulate</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QComboBox" name="selectobject">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>80</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Sphere</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Cube</string>
     </property>
    </item>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>20</y>
      <width>821</width>
      <height>601</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2"/>
   </widget>
   <widget class="QLineEdit" name="lineEdit_2">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>360</y>
      <width>71</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>0.2</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>340</y>
      <width>161</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Friction Coefficient</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_3">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>410</y>
      <width>71</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>0.5</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>410</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Bouncing Coefficient</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>260</y>
      <width>67</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Gravity</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_4">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>200</y>
      <width>113</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>100</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>130</y>
      <width>181</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Rigid Body Simualtion</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>50</y>
      <width>221</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Please select particle type here.</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>440</y>
      <width>281</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>All the parameters are changable </string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>470</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>at run time.</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>0</y>
      <width>281</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Try default settings first is highly </string>
    </property>
    <property name="wordWrap">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>30</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>recommended.</string>
    </property>
   </widget>
   <zorder>layoutWidget</zorder>
   <zorder>line</zorder>
   <zorder>label</zorder>
   <zorder>lineEdit</zorder>
   <zorder>checkBox_2</zorder>
   <zorder>checkBox_4</zorder>
   <zorder>selectobject</zorder>
   <zorder>horizontalLayoutWidget</zorder>
   <zorder>lineEdit_2</zorder>
   <zorder>label_2</zorder>
   <zorder>lineEdit_3</zorder>
   <zorder>label_3</zorder>
   <zorder>label_4</zorder>
   <zorder>lineEdit_4</zorder>
   <zorder>checkBox</zorder>
   <zorder>label_5</zorder>
   <zorder>label_6</zorder>
   <zorder>label_7</zorder>
   <zorder>label_8</zorder>
   <zorder>label_9</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1175</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMain_Window">
    <property name="title">
     <string>Welcome</string>
    </property>
   </widget>
   <addaction name="menuMain_Window"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>StartSimulate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>53</x>
     <y>562</y>
    </hint>
    <hint type="destinationlabel">
     <x>76</x>
     <y>509</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>selectobject</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>Choose_ParticleType()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>115</x>
     <y>97</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>Set_Gravity()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>242</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>Gravity_change_flag()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>182</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_2</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>Add_GroundPlane()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>69</x>
     <y>338</y>
    </hint>
    <hint type="destinationlabel">
     <x>23</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>564</y>
    </hint>
    <hint type="destinationlabel">
     <x>207</x>
     <y>648</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_4</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>Set_Particle_Count()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>231</x>
     <y>232</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Set_Cloth_Sim()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>16</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Set_Rigid_Sim()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>167</y>
    </hint>
    <hint type="destinationlabel">
     <x>22</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_2</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>set_Friction_Coefficient()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>230</x>
     <y>400</y>
    </hint>
    <hint type="destinationlabel">
     <x>264</x>
     <y>385</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_3</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>set_Bouncing_Coefficient()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>247</x>
     <y>454</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>463</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>StartSimulate()</slot>
  <slot>Choose_ParticleType()</slot>
  <slot>Set_Gravity()</slot>
  <slot>Gravity_change_flag()</slot>
  <slot>Add_GroundPlane()</slot>
  <slot>Reset()</slot>
  <slot>Set_Particle_Count()</slot>
  <slot>Set_Cloth_Sim()</slot>
  <slot>Set_Rigid_Sim()</slot>
  <slot>set_Friction_Coefficient()</slot>
  <slot>set_Bouncing_Coefficient()</slot>
 </slots>
</ui>
